source/renderer/app/stores/ada/AdaWalletsStore.js

```diff
@@ -186,6 +245,162 @@ export default class AdaWalletsStore extends WalletStore {
     }
   };
 
+  _generateCertificate = async (params: {
+    filePath: string,
+  }) => {
+    try {
+      // Stop polling
+      this.actions.networkStatus.stopPoller.trigger();
+
+      // Set inProgress state to show spinner if is needed
+      this._updateCertificateCreationState(true);
+
+      // Genereate 12-word mnemonic
+      const recoveryPhrase: GetWalletRecoveryPhraseResponse = await (
+        this.getWalletRecoveryPhraseRequest.execute().promise
+      );
+
+      // Generate 9-words (additional) mnemonic
+      const additionalMnemonicWords: GetWalletCertificateAdditionalMnemonicsResponse = await (
+        this.getWalletCertificateAdditionalMnemonicsRequest.execute().promise
+      );
+      this.additionalMnemonicWords = additionalMnemonicWords.join(' ');
+
+      // Generate spending password from 9-word mnemonic and save to store
+      const spendingPassword = mnemonicToSeedHex(this.additionalMnemonicWords);
+      this.walletCertificatePassword = spendingPassword;
+
+      // Generate paper wallet scrambled mnemonic
+      const walletCertificateRecoveryPhrase: ?GetWalletCertificateRecoveryPhraseResponse = await (
+        this.getWalletCertificateRecoveryPhraseRequest.execute({
+          passphrase: spendingPassword,
+          input: recoveryPhrase.join(' '),
+        }).promise
+      );
+      this.walletCertificateRecoveryPhrase = walletCertificateRecoveryPhrase
+        ? walletCertificateRecoveryPhrase.join(' ')
+        : '';
+
+      // Create temporary wallet
+      const walletData = {
+        name: 'Paper Wallet',
+        mnemonic: recoveryPhrase.join(' '),
+        password: spendingPassword,
+      };
+      const wallet = await this.createWalletRequest.execute(walletData).promise;
+
+      // Get temporary wallet address
+      let walletAddresses;
+      if (wallet) {
+        walletAddresses = await this.getWalletAddressesRequest.execute({
+          walletId: wallet.id,
+        }).promise;
+
+        // delete temporary wallet
+        await this.deleteWalletRequest.execute({ walletId: wallet.id });
+      }
+
+      // Set wallet certificate address
+      let walletAddress;
+      if (walletAddresses) {
+        walletAddress = get(walletAddresses, ['addresses', '0', 'id'], null);
+        this.walletCertificateAddress = walletAddress;
+      }
+
+      // download pdf certificate
+      this._downloadCertificate(
+        walletAddress,
+        walletCertificateRecoveryPhrase,
+        params.filePath,
+      );
+    } catch (error) {
+      throw error;
+    } finally {
+      this.actions.networkStatus.restartPoller.trigger();
+    }
+  };
+
+  _downloadCertificate = action((
+    address: string,
+    recoveryPhrase: Array<string>,
+    filePath: string,
+  ) => {
+    setTimeout(() => { // Timeout is used to allow enought time for button text re-rendering
```

Comment:
@tomothespian I don't think we need this timeout anymore. Please remove it and see if it works as expected.